feat: add sign-schnorr command to etosd CLI (#309)
babylonchain/finality-provider: docs/eots.md

- **Linux** `~/.Eotsd`
- **Windows** `C:\Users\<username>\AppData\Local\Eotsd`

## 3. Keys Management

Handles the keys for EOTS.

> ### 3.1. Create EOTS Key

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/7a14a12e5bc4b028c97e57d8751f85f59b246cee

Enter keyring passphrase (attempt 1/3):
...

2024-04-25T17:11:09.369163Z     info    successfully created an EOTS key        {"key name": "my-key-name", "pk": "50b106208c921b5e8a1c45494306fe1fc2cf68f33b8996420867dc7667fde383"}
New key for the BTC chain is created (mnemonic should be kept in a safe place for recovery):
{
  "name": "my-key-name",
  "pub_key_hex": "50b106208c921b5e8a1c45494306fe1fc2cf68f33b8996420867dc7667fde383",
  "mnemonic": "bad mnemonic private tilt wish bulb miss plate achieve manage feel word safe dash vanish little miss hockey connect tail certain spread urban series"
}
```

> Store the mnemonic in a safe place. With the mnemonic only it is possible to
recover the generated keys by using the `--recover` flag.

> ### 3.2. Recover Key

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/7a14a12e5bc4b028c97e57d8751f85f59b246cee

To recover the keys from a mnemonic, run:

```shell
eotsd keys add --home /path/to/eotsd/home/ --key-name my-key-name --keyring-backend file --recover

> Enter your mnemonic
bad mnemonic private tilt wish bulb miss plate achieve manage feel word safe dash vanish little miss hockey connect tail certain spread urban series
2024-04-25T17:13:29.681324Z     info    successfully created an EOTS key        {"key name": "my-key-name", "pk": "50b106208c921b5e8a1c45494306fe1fc2cf68f33b8996420867dc7667fde383"}
New key for the BTC chain is created (mnemonic should be kept in a safe place for recovery):
{
  "name": "my-key-name",
  "pub_key_hex": "50b106208c921b5e8a1c45494306fe1fc2cf68f33b8996420867dc7667fde383",
  "mnemonic": "noise measure tuition inform battle swallow slender bundle horn pigeon wage mule average bicycle claim solve home swamp banner idle chapter surround edit gossip"
}
```

You will be prompt to provide the mnemonic on key creation. The generated keys

> should be exactly the same.

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/7a14a12e5bc4b028c97e57d8751f85f59b246cee

> ### 3.3. Option to Sign Schnorr

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/7a14a12e5bc4b028c97e57d8751f85f59b246cee

> You can use your key to create a Schnorr signature over arbitrary data,

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/7a14a12e5bc4b028c97e57d8751f85f59b246cee

  "mnemonic": "bad mnemonic private tilt wish bulb miss plate achieve manage feel word safe dash vanish little miss hockey connect tail certain spread urban series"
}
```

> ## 5. Sign Schnorr

Reviewer: Maybe we can rename "Create EOTS Keys" to "Keys Management" and make this a subsection there called "Signing Messages".

Author: I have made it into 3 subsections https://github.com/babylonchain/finality-provider/pull/309/commits/567c27d9db8e048ad496903d70d7c02db104f21a

- ## 3. Keys Management
  - ### 3.1. Create EOTS Key
  - ### 3.2. Recover Key
  - ### 3.3. Option to Sign Schnorr
  
Let me know what you think

> If you have the key created on the keyring and want Schnorr signature over
> arbitrary data, there is an option to do it by running `eotsd sign-schnorr`.

Reviewer: ```suggestion
You can use your key to create a Schnorr signature over arbitrary data,
through the `eotsd sign-schnorr` command.
```

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/90710507644ed6e4443a5efbb234e5413f23dd90

  "mnemonic": "bad mnemonic private tilt wish bulb miss plate achieve manage feel word safe dash vanish little miss hockey connect tail certain spread urban series"
}
```

## 5. Sign Schnorr

If you have the key created on the keyring and wants to schnorr sign any data,
there is an option to do it by running `eotsd sign-schnorr`. The command takes
as a argument the file path which should contain the msg that will be hashed by
sha256 to reduce the size to 32 bytes and sign it using the private BTC key

> associated with the `--key-name` flag.

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/988b1ee798359d01639716429e9b1aa8c5e445e7

```shell

> eotsd sign-schnorr /path/to/msg.x --home /path/to/eotsd/home/ --key-name my-key-name

Author: done at https://github.com/babylonchain/finality-provider/pull/309/commits/988b1ee798359d01639716429e9b1aa8c5e445e7

{
  "key_name": "my-key-name",
  "pub_key_hex": "667ebbe6cec5889d20ab56e05a4c1402bf9369637ae2e2ec26c12fe02405a22b",
  "signed_msg_hex": "b123ef5f69545cd07ad505c6d3b4931aa87b6adb361fb492275bb81374d98953",

>   "serialized_signature_hex": "36b23d0ecd40a5da16ca2ef3550a364583f5abab69a9b4eae63199a3f98f0d1442fffe9c4f53fe345990e777361db94a53d4db3d68c3d2d7bbd2ca28901288cc"

Reviewer: Don't see this update

>   "serialized_signature_hex": "36b23d0ecd40a5da16ca2ef3550a364583f5abab69a9b4eae63199a3f98f0d1442fffe9c4f53fe345990e777361db94a53d4db3d68c3d2d7bbd2ca28901288cc"

Author: https://github.com/babylonchain/finality-provider/pull/309/commits/5bafb1bc91eb1fe86383a34463c0147637026850
My bad, not sure what happend with my commit change :thinking:
